src/test/java/core/basesyntax/HelloWorldTest.java

```diff
@@ -1,8 +1,184 @@
 package core.basesyntax;
 
+import core.basesyntax.model.User;
+import core.basesyntax.service.ImpossibleRegisterUserException;
+import core.basesyntax.service.RegistrationServiceImpl;
+import org.junit.Assert;
+import org.junit.jupiter.api.Assertions;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+
+
 /**
  * Feel free to remove this class and create your own.
  */
 public class HelloWorldTest {
+    private RegistrationServiceImpl registrationService = new RegistrationServiceImpl();
+    private User user = new User();
+
+    @BeforeEach
+    void setUp() {
+        user = new User();
+    }
+
+    @Test
+    void loginOfUserAlreadyExists_NotOk() {
+        user.setId(12l);
+        user.setLogin("sanya67");
+        user.setAge(18);
+        user.setPassword("123456");
+        try {
+            Assert.assertEquals(user, registrationService.register(user));
+        } catch (ImpossibleRegisterUserException e) {
+            System.out.println("this login isn`t available");
+        }
+        User user1 = new User();
+        user.setId(12l);
+        user.setLogin("sanya67");
+        user.setAge(18);
+        user.setPassword("123456");
+        Assertions.assertThrows(ImpossibleRegisterUserException.class,
+                ()-> {
+                    registrationService.register(user1);
+                });
+    }
+
+    @Test
+    void loginIsNull_NotOk() {
+        user.setLogin(null);
+        user.setId(11l);
+        user.setPassword("890765");
+        user.setAge(20);
+        Assertions.assertThrows(ImpossibleRegisterUserException.class,
+                ()-> {
+            registrationService.register(user);
+                });
+    }
+
+    @Test
+    void loginNotExist_Ok() {
+        user.setId(8l);
+        user.setLogin("Deadpool");
+        user.setAge(89);
+        user.setPassword("90809809");
+        try {
+            Assert.assertEquals(user, registrationService.register(user));
+        } catch (ImpossibleRegisterUserException e) {
+            System.out.println("Can`t register this user");
+        }
```

Comment:
Similar to the previous comment, using `Assertions.assertThrows` would be more appropriate here to check for exceptions instead of using a try-catch block.